assets/js/romo/currency_text_input.js

```diff
@@ -0,0 +1,168 @@
+      this.elem.data('romo-currency-text-input-indicator-padding-px')
+    );
+  }
+  if (this.elem.data('romo-currency-text-input-elem-display') !== undefined) {
+    this.elem.attr(
+      'data-romo-indicator-text-input-elem-display',
+      this.elem.data('romo-currency-text-input-elem-display')
+    );
+  }
+  if (this.elem.data('romo-currency-text-input-btn-group') !== undefined) {
+    this.elem.attr(
+      'data-romo-indicator-text-input-btn-group',
+      this.elem.data('romo-currency-text-input-btn-group')
+    );
+  }
+
+  this.elem.on('indicatorTextInput:indicatorClick', $.proxy(function(e) {
+    this.elem.trigger('romoCurrencyTextInput:indicatorClick', []);
+  }, this));
+
+  this.elem.on('romoCurrencyTextInput:triggerPlaceIndicator', $.proxy(function(e) {
+    this.elem.trigger('indicatorTextInput:triggerPlaceIndicator', []);
+  }, this));
+  this.elem.on('romoCurrencyTextInput:triggerEnable', $.proxy(function(e) {
+    this.elem.trigger('indicatorTextInput:triggerEnable', []);
+  }, this));
+  this.elem.on('romoCurrencyTextInput:triggerDisable', $.proxy(function(e) {
+    this.elem.trigger('indicatorTextInput:triggerDisable', []);
+  }, this));
+  this.elem.on('romoCurrencyTextInput:triggerShow', $.proxy(function(e) {
+    this.elem.trigger('indicatorTextInput:triggerShow', []);
+  }, this));
+  this.elem.on('romoCurrencyTextInput:triggerHide', $.proxy(function(e) {
+    this.elem.trigger('indicatorTextInput:triggerHide', []);
+  }, this));
+
+  this.elem.romoIndicatorTextInput();
+}
+
+// private
+
+RomoCurrencyTextInput.prototype._formatCurrencyValue = function(sanitizedValue) {
+  return RomoCurrency.format(sanitizedValue, {
+    'thousandsDelim':   this._getFormatThousandsDelim(),
+    'decimalDelim':     this._getFormatDecimalDelim(),
+    'numDecimalPlaces': this._getFormatNumDecimalPlaces()
+  });
+}
+
+RomoCurrencyTextInput.prototype._unFormatCurrencyValue = function(inputValue) {
+  return RomoCurrency.unFormat(inputValue, {
+    'numDecimalPlaces': this._getFormatNumDecimalPlaces()
+  });
+}
+
+RomoCurrencyTextInput.prototype._getFormatThousandsDelim = function() {
+  return (
+    this.elem.data('romo-currency-text-input-format-thousands-delim') ||
+    this.defaultFormatThousandsDelim
+  );
+}
+
+RomoCurrencyTextInput.prototype._getFormatDecimalDelim = function() {
+  return (
+    this.elem.data('romo-currency-text-input-format-decimal-delim') ||
+    this.defaultFormatDecimalDelim
+  );
+}
+
+RomoCurrencyTextInput.prototype._getFormatNumDecimalPlaces = function() {
+  var places = this.elem.data('romo-currency-text-input-format-num-decimal-places');
+  return !isNaN(places = Math.abs(places)) ? places : this.defaultFormatNumDecimalPlaces;
+}
+
+RomoCurrencyTextInput.prototype._getHiddenInputElem = function() {
+  return $('<input type="hidden" name="'+this.elem.attr('name')+'" value="'+this.elem.val()+'">');;
+}
+
+RomoCurrencyTextInput.prototype._getInputNameSuffix = function() {
```

Comment:
@kellyredding - You might call this `getInputName` and drop the suffix. I think this is returning the full input name.